pkg/bu_server/storage/postgres/trade_document.go

```diff
@@ -58,7 +80,12 @@ WITH latest_visible AS (
 		NOT (meta ? 'deleted') AND
 		($3 = 0 OR $3 = kind) AND
 		(COALESCE(ARRAY_LENGTH($4::TEXT[], 1), 0) = 0 OR doc_id = ANY($4)) AND
-		($5::JSONB IS NULL OR meta @> $5)
+		($5::JSONB IS NULL OR meta @> $5) AND
+		(
+			($8 = '' AND $9 = '') OR
+			meta->>'from' = ANY(SELECT * FROM filtered_bu) OR
```

Comment:
this statement might suffer from performance issue in the future while the table getting larger. it's not an issue for now.